vm/src/stdlib/itertools.rs

```diff
@@ -1264,7 +1266,42 @@ mod decl {
     }
 
     #[pyimpl(with(IterNext, Constructor))]
-    impl PyItertoolsCombinations {}
+    impl PyItertoolsCombinations {
+        #[pymethod(magic)]
+        fn reduce(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyTupleRef {
+            if zelf.result.read().is_none() {
```

Comment:
zelf.result.read() is repeated here and below.

Because here is no other locking chance, reusing will be better.

```rust
if let Some(result) = zelf.result.read() {
    ...  // is_some case (second and third block) here
} else {
    ...  // is_none case (first block) here
}
```